torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1706,84 +1750,181 @@ struct to_ir {
   }
 
   CondValue emitIsInstance(const Expr& obj, const Expr& classinfo) {
-    // turn (float, (int, tuple)) into a flat list of types and type kind
-    // category checks: tuple_check = true, types = {float, int}
-    struct GatheredTypes {
-      GatheredTypes(ScriptTypeParser parser) : typeParser_(std::move(parser)) {}
-      void gather(const Expr& classinfo) {
-        if (classinfo.kind() == TK_TUPLE_LITERAL) {
-          for (Expr e : TupleLiteral(classinfo).inputs()) {
-            gather(e);
+    // For use with `unifyTypeList`
+    std::stringstream nowhere;
```

Comment:
can you declare this closer to where it is used ? it's a little confusing to see it here. or, make unifyTypeList take in an optional `why_not` parameter